src/converter/one/convert/JfrToOtlp.java

```diff
@@ -0,0 +1,271 @@
+
+    private final Proto proto = new Proto(1024);
+    private final double ticksPerNanosecond;
+
+    public JfrToOtlp(JfrReader jfr, Arguments args) {
+        super(jfr, args);
+        ticksPerNanosecond = jfr.ticksPerSec / 1_000_000_000.0;
+    }
+
+    @Override
+    public void convert() throws IOException {
+        long rpMark = proto.startField(PROFILES_DATA_resource_profiles, MSG_LARGE);
+        long spMark = proto.startField(RESOURCE_PROFILES_scope_profiles, MSG_LARGE);
+        super.convert();
+        proto.commitField(spMark);
+        proto.commitField(rpMark);
+
+        writeProfileDictionary();
+    }
+
+    @Override
+    protected void convertChunk() {
+        long pMark = proto.startField(SCOPE_PROFILES_profiles, MSG_LARGE);
+
+        writeSampleTypes();
+        writeTimingInformation();
+
+        List<Integer> locationIndices = new ArrayList<>();
+        collector.forEach(new OtlpEventToSampleVisitor(locationIndices));
+
+        long liMark = proto.startField(PROFILE_location_indices, MSG_LARGE);
+        locationIndices.forEach(proto::writeInt);
+        proto.commitField(liMark);
+
+        proto.commitField(pMark);
+    }
+
+    private void writeSampleTypes() {
+        long stsMark = proto.startField(PROFILE_sample_type, MSG_SMALL);
+        proto.field(VALUE_TYPE_type_strindex, stringPool.index(getValueType()));
+        proto.field(VALUE_TYPE_unit_strindex, stringPool.index(getSampleUnits()));
+        proto.field(VALUE_TYPE_aggregation_temporality, AGGREGATION_TEMPORARALITY_cumulative);
+        proto.commitField(stsMark);
+
+        long sttMark = proto.startField(PROFILE_sample_type, MSG_SMALL);
+        proto.field(VALUE_TYPE_type_strindex, stringPool.index(getValueType()));
+        proto.field(VALUE_TYPE_unit_strindex, stringPool.index(getTotalUnits()));
+        proto.field(VALUE_TYPE_aggregation_temporality, AGGREGATION_TEMPORARALITY_cumulative);
+        proto.commitField(sttMark);
+    }
+
+    private void writeTimingInformation() {
+        proto.field(PROFILE_time_nanos, jfr.chunkStartNanos);
+        proto.field(PROFILE_duration_nanos, jfr.chunkDurationNanos());
+    }
+
+    public void dump(OutputStream out) throws IOException {
+        out.write(proto.buffer(), 0, proto.size());
+    }
+
+    private void writeProfileDictionary() {
+        long profilesDictionaryMark = proto.startField(PROFILES_DATA_dictionary, MSG_LARGE);
+
+        // Mapping[0] must be a default mapping according to the spec
+        long mMark = proto.startField(PROFILES_DICTIONARY_mapping_table, MSG_SMALL);
+        proto.commitField(mMark);
+
+        // Write function table
+        for (String name : functionPool.keys()) {
+            long fMark = proto.startField(PROFILES_DICTIONARY_function_table, MSG_SMALL);
+            proto.field(FUNCTION_name_strindex, stringPool.index(name));
+            proto.commitField(fMark);
+        }
+
+        // Write location table
+        for (Line l : linePool.keys()) {
```

Comment:
You took my suggestion about shortening variable names too literally :)
`line` would be perfect here. `l` is not good as identifier - it resembles `1` or `I`.